components/infobox/wikis/starcraft2/infobox_person_custom_shared.lua

```diff
@@ -0,0 +1,176 @@
+---
+-- @Liquipedia
+-- wiki=starcraft2
+-- page=Module:Infobox/Person/Custom/Shared
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+--Shared functions between SC2 person infoboxes (Player/Commentator and MapMaker)
+
+local String = require('Module:StringUtils')
+local Variables = require('Module:Variables')
+local RaceIcon = require('Module:RaceIcon').getBigIcon
+local CleanRace = require('Module:CleanRace')
+local Logic = require('Module:Logic')
+
+--race stuff tables
+local _FACTION1 = {
+	['p'] = 'Protoss', ['pt'] = 'Protoss', ['pz'] = 'Protoss',
+	['t'] = 'Terran', ['tp'] = 'Terran', ['tz'] = 'Terran',
+	['z'] = 'Zerg', ['zt'] = 'Zerg', ['zp'] = 'Zerg',
+	['r'] = 'Random', ['a'] = 'All'
+}
+local _FACTION2 = {
+	['pt'] = 'Terran', ['pz'] = 'Zerg',
+	['tp'] = 'Protoss', ['tz'] = 'Zerg',
+	['zt'] = 'Terran', ['zp'] = 'Protoss'
+}
+
+--role stuff tables
+local _ROLES = {
+	['admin'] = 'Admin', ['analyst'] = 'Analyst', ['coach'] = 'Coach',
+	['commentator'] = 'Commentator', ['caster'] = 'Commentator',
+	['expert'] = 'Analyst', ['host'] = 'Host', ['streamer'] = 'Streamer',
+	['interviewer'] = 'Interviewer', ['journalist'] = 'Journalist',
+	['manager'] = 'Manager', ['player'] = 'Player',
+	['map maker'] = 'Map maker', ['mapmaker'] = 'Map maker',
+	['observer'] = 'Observer', ['photographer'] = 'Photographer',
+	['tournament organizer'] = 'Organizer', ['organizer'] = 'Organizer',
+}
+local _CLEAN_OTHER_ROLES = {
+	['blizzard'] = 'Blizzard', ['coach'] = 'Coach', ['staff'] = 'false',
+	['content producer'] = 'Content producer', ['streamer'] = 'false',
+}
+
+local _raceData
+local _statusStore
+local _militaryStore
+local _args
+
+local _PAGENAME = mw.title.getCurrentTitle().prefixedText
+
+local CustomPerson = {}
+
+function CustomPerson.setArgs(args)
+	_args = args
+end
+
+function CustomPerson.shouldStoreData()
+	if
+		_args.disable_smw == 'true' or _args.disable_lpdb == 'true' or _args.disable_storage == 'true'
+		or Variables.varDefault('disable_SMW_storage', 'false') == 'true'
+		or mw.title.getCurrentTitle().nsText ~= ''
+	then
+		Variables.varDefine('disable_SMW_storage', 'true')
+		return false
+	end
+	return true
+end
+
+function CustomPerson.nameDisplay()
+	CustomPerson.getRaceData(_args.race or 'unknown')
+	local raceIcon = RaceIcon({'alt_' .. _raceData.race})
+	local name = _args.id or _PAGENAME
+
+	return raceIcon .. '&nbsp;' .. name
+end
+
+function CustomPerson.getRaceData(race)
+	race = string.lower(race)
+	race = CleanRace[race] or race
+	local display = CustomPerson.raceDisplayLookupTable[race]
+	if not display and race ~= 'unknown' then
+		display = '[[Category:InfoboxRaceError]]<strong class="error">' ..
+			mw.text.nowiki('Error: Invalid Race') .. '</strong>'
+	end
+
+	_raceData = {
+		race = race,
+		faction = _FACTION1[race] or '',
+		faction2 = _FACTION2[race] or '',
```

Comment:
```suggestion
		faction = _FACTIONS[race:sub(1,1)] or '',
		faction2 = _FACTIONS[race:sub(2,2)] or '',
```

Reply:
i thought about doing it that way, but this would allow crap input to be (wrongly) recognized as proper input
if one e.g. would input `qt` it would still set `faction2 = Terran` which is not intended
and it would allow `All` and `Random` as valid input for `faction2` which is also not intended

Reply:
used a different way to circumvent it and also remove the tables (now using the table i added for the display)